cake-autorate.sh

```diff
@@ -346,10 +347,23 @@ maintain_log_file()
 				((log_file_size_KB=log_file_size_bytes/1024))
 				log_msg "DEBUG" "log file size: ${log_file_size_KB} KB has exceeded configured maximum: ${log_file_max_size_KB} KB so flushing and rotating log file."
 				break
-			elif (( reset_log_file_signalled ))
```

Comment:
nitpicking but this is very awkward, instead of `elif (( kill_maintain_log_file_signalled || export_log_file_signalled || reset_log_file_signalled ))`... do:

```shell
				elif ((kill_maintain_log_file_signalled))
				then
					flush_log_pipe
					trap - TERM EXIT
					exit
				elif ((export_log_file_signalled))
				then
					log_msg "DEBUG" "received log file export signal so exporting log file."
					export_log_file
					export_log_file_signalled=0
				elif ((reset_log_file_signalled))
				then
					log_msg "DEBUG" "received log file reset signal so flushing and resetting log file."
					break
				fi
```

Reply:
I thought about that, but surely that would be a lot slower?

Reply:
I don't know if opening a new arithmetic context is slower, but if you think it's slower then maybe you should avoid it in general and stick to `test`